doc/mdc-migration.md

```diff
@@ -0,0 +1,54 @@
+# MDC Android migration
+## Status:
+Accepted on 2020-04-16
+
+## Context:
+We have been using app compat design theme from the start, while that worked perfectly it did not allow us to use new material components because those are dependent on `Theme.MaterialComponents.*` app theme. Using new material components allows us to have much more control over component theming like text styles, icon & icon gravity, strokes, corners, etc.,
+
+- Primary reason why we are moving to Material Design Components theme is to use `MaterialButton`,  and avoid using different buttons such as `Button` , `OmegaCenterIconButton` or `PrimarySolidButton`.  The material theme migration will also allows us to use other material components. You can find more components supported by Android and how to theme them [here](https://material.io/components).
+- This migration will also open better theming support for material components such as corner radius, strokes, shapes etc.,
+- You can find more information on setting up MDC Android [here](https://material.io/develop/android/docs/getting-started/).
+
+## Changes:
+### Button tag change:
+- One of the primary change with this migration is using `Button` tag, we will start using `com.google.android.material.button.MaterialButton` in views instead of `Button`, `OmegaCenterIconButton`, `PrimarySolidButton`.
+- In order to use icon in a `MaterialButton` use the `icon` attribute instead of `drawableStart` or `drawableEnd` attr.
+- `MaterialButton` also allows us to set `iconGravity` (`textStart`, `viewStart`, `textEnd`, `viewEnd`) for setting the icon position, `iconTint` for setting the icon color.
+  - By default `Clinic.V2.MaterialButton` & other primary button styles for Simple use `iconGravity` of `textStart`, if you want to change that create a new style or override it in the layout file.
+- Material theme will automatically convert any `Button` to `MaterialButton` internally using custom inflater. So if there is a view that extends button the theme values may need to changed if the design specs are broken.
+
```

Comment:
We need to derive the gist of how moving to this will improve our code quality and ease of use. We do not need the implementation details as much in an ADR.